File: Palindrome.py
# Pake string
def isPalindrome(x: int) -> bool:
	xs = str(x)
	nsx = xs[::-1]
	return xs == nsx

# print(isPalindrome(121))


# Tanpa di convert ke string
def isPalindrome(x: int) -> bool:
	if x < 0:
		return False

	reverse = 0
	num = x

	while num:
		lastdigit = num % 10 # 1
		reverse = reverse * 10 + lastdigit #1
		num = num // 10 #12

	hasil = reverse == x
	print(f"\nHasil = {reverse} || {x} -> {hasil}\n")

isPalindrome(int(input("Masukkan Angka: ")))

File: Palindrome.cpp
#include <iostream>
using namespace std;

bool isPalindrome(int x) {
    if (x < 0) {
        return false;
    }

    int reverse = 0;
    int num = x;

    while (num) {
        int lastdigit = num % 10;
        reverse = reverse * 10 + lastdigit;
        num = num / 10;
    }

    bool hasil = (reverse == x);
    cout << "\nHasil = " << reverse << " || " << x << " -> " << hasil << "\n" << endl;
    return hasil;
}

int main() {
    int x;
    cout << "Masukkan Angka: ";
    cin >> x;
    isPalindrome(x);
    return 0;
}
